src/path.cr

```diff
@@ -190,24 +189,22 @@ struct Path
       reader.previous_char
     end
 
+    if reader.pos == 0 && !separators.includes?(reader.current_char)
+      if windows? && reader.has_next? && reader.peek_next_char == ':'
+        reader.next_char
```

Comment:
At this point we can actually check for `self.anchor` and return its value. The conditional in lines 210-212 can probably be  removed then.

Reply:
So it can be like this. But lines 210-212 cannot be removed, it breaks tests for C:\foo etc.
And IMHO your example for Path#parents is correct, i don't see error.